### PR TITLE
Adds a clown hardsuit

### PR DESCRIPTION
**Pull request by Militaires**
[Changelogs]: # (Please make a changelog if you're adding, removing or changing content that'll affect players. This includes, but is not limited to, new features, sprites, sounds; balance changes; map edits and important fixes)
[]: # (See here for how to easily make a changelog: https://github.com/tgstation/tgstation/wiki/Changelogs. An example changelog has been provided below. Please edit or remove)

:cl: Basilman
add: Added an admin-spawnable only "Cosmohonk" hardsuit that has the same protection values of an engineering hardsuit. Comes with built-in lights and requires clown shoes on to be equipped.
/:cl:

edit: fuck forgot sprites bear with me
cosmohonkunworn
honksuitunworn - copy
honksuit0north
honksuit0 - copy
honksuit0
honksuit0east
cosmohonk1
cosmohonk
cosmohonk0

features :-

- inbuilt hardsuit lights for doing crazy pranks even if the station is dark and depressurized.
- two suit storage units for keeping your most beloved possessions (your bike horns) safe.
- revolutionary advanced smooth bananium plating slips any attacks made to you off you! and is even capable of slipping a colliding meteor*.
- logitech honk-series electronics for comfort and utility.
- logitech honk-series clown shoes that are colourful as hell!

### CODE REVIEW

File: code/modules/clothing/spacesuits/hardsuit.dm [hunk 1]
```diff
@@ -494,6 +494,34 @@
...
+	desc = "A special suit that protects against hazardous, low humor environments. Has radiation shielding. Requires you to wear clown shoes to put on."
+	icon_state = "hardsuit-clown"
+	item_state = "clown_hardsuit"
+	armor = list(melee = 30, bullet = 5, laser = 10, energy = 5, bomb = 10, bio = 100, rad = 75)
+	helmettype = /obj/item/clothing/head/helmet/space/hardsuit/clown
+	pockets = /obj/item/weapon/storage/internal/pocket
+
+/obj/item/clothing/suit/space/hardsuit/clown/mob_can_equip(mob/M, slot)
+	if(!ishuman(M))
+		return 0
```
Comment (optimumtact) [inline code/modules/clothing/spacesuits/hardsuit.dm]: use the FALSE define here

    return FALSE

same below where you return 1, use the true define

     return TRUE

**Comment by optimumtact** [comment]
Also please keep the title of your PR brief and to the point.

**Comment by Militaires** [comment]
@optimumtact  but i like diahrea titles it's a motherfucking art, o well, also thanks for the advice, i'll commit it rn.

**Comment by Militaires** [comment]
also big thanks to @AnturK  for bearing with me being autistic.

File: code/modules/clothing/spacesuits/hardsuit.dm [hunk 2]
```diff
@@ -494,6 +494,32 @@
...
+	item_state = "clown_hardsuit"
+	armor = list(melee = 30, bullet = 5, laser = 10, energy = 5, bomb = 10, bio = 100, rad = 75)
+	helmettype = /obj/item/clothing/head/helmet/space/hardsuit/clown
+	pockets = /obj/item/weapon/storage/internal/pocket
+
+/obj/item/clothing/suit/space/hardsuit/clown/mob_can_equip(mob/M, slot)
+	if(!..() || !ishuman(M))
+		return FALSE
+	var/mob/living/carbon/human/H = M
+	if(istype(H.shoes,/obj/item/clothing/shoes/clown_shoes))
```
Comment (KorPhaeron) [inline code/modules/clothing/spacesuits/hardsuit.dm]: Can't I just put on clown shoes, put this on, then take the clown shoes off? What's the point of the check?

Comment (hornygranny) [inline code/modules/clothing/spacesuits/hardsuit.dm]: if you want to limit this to clowns check for the clown special role

Comment (Militaires) [inline code/modules/clothing/spacesuits/hardsuit.dm]: @hornygranny  @KorPhaeron  true, ill make it limit to clown check or clown ID like the H.O.N.K

**Comment by optimumtact** [comment]
You need to make sure this compiles before I can merge it.

File: code/modules/clothing/spacesuits/hardsuit.dm [hunk 3]
```diff
@@ -494,6 +494,32 @@
...
+	item_state = "clown_hardsuit"
+	armor = list(melee = 30, bullet = 5, laser = 10, energy = 5, bomb = 10, bio = 100, rad = 75)
+	helmettype = /obj/item/clothing/head/helmet/space/hardsuit/clown
+	pockets = /obj/item/weapon/storage/internal/pocket
+
+/obj/item/clothing/suit/space/hardsuit/clown/mob_can_equip(mob/M, slot)
+	if(!..() || !ishuman(M))
+		return FALSE
+	var/mob/living/carbon/human/H = M
+	if(M.mind.assigned_role == "Clown")
```
Comment (PKPenguin321) [inline code/modules/clothing/spacesuits/hardsuit.dm]: replace the M on this line with H

Comment (PKPenguin321) [inline code/modules/clothing/spacesuits/hardsuit.dm]: @Militaires just merge this PR i've saved you the effort of even needing to do this yourself https://github.com/Militaires/tgstation/pull/4

**Comment by optimumtact** [comment]
Open a new pr or ping me on this one when you're ready to continue.

**Pull request closed without merging**